catkin_make_repo

```diff
@@ -11,7 +11,14 @@ RUN_TESTS=$1
 NUM_PARALLEL_TEST_JOBS=$2
 PACKAGES=${*:3}
 
-MAKE_ARGS="-j$((`nproc`+1)) VERBOSE=1"
+if [[ "$OSTYPE" == "linux-gnu" ]]; then
+  MAKE_ARGS="-j$((`nproc`+1)) VERBOSE=1"
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+  MAKE_ARGS="-j$((`sysctl -n hw.ncpu`+1)) VERBOSE=1"
```

Comment:
I would suggest an intermediate variable containing only the number of hardware threads, `n`, or `nhts` and then use it once to compute the value for `MAKE_ARGS` to avoid duplication.
This could include the + 1, but  actually I would be careful to count HT threads as full threads on any machine. I expect it to be slower than with using let's say 3/4 of it. But I never tested this.